### PR TITLE
apm: add python tab to connect_dbm_and_apm docs

### PR DESCRIPTION
**Pull request by mabdinur**
### What does this PR do?

Documents python support for DBM-APM linking. 

### Motivation

Document new feature.

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/database_monitoring/guide/connect_dbm_and_apm.md [hunk 1]
```diff
@@ -144,6 +144,53 @@ client.query("SELECT 1;")
...
+pip install psycopg2
+```
+
+Enable the database monitoring propagation feature by setting the following enviornment variable:
+   `DD_TRACE_SQL_COMMENT_INJECTION_MODE=full`
+
+For the best user experience ensure the following enviornemnts variables are set in your application:
+   `DD_SERVICE=(application name)`
+   `DD_ENV=(application enviornment)`
+   `DD_VERSION=(application version)`
```
Comment (mabdinur) [inline content/en/database_monitoring/guide/connect_dbm_and_apm.md:165]:
```suggestion
   - `DD_SERVICE=(application name)`
   - `DD_ENV=(application enviornment)`
   - `DD_VERSION=(application version)`
```

File: content/en/database_monitoring/guide/connect_dbm_and_apm.md [hunk 2]
```diff
@@ -144,6 +144,53 @@ client.query("SELECT 1;")
...
+pip install "ddtrace>=1.7.0"
+```
+
+Install [psycopg2][2] (Note - DBM-APM linking is not currently supported for mysql clients):
+```
+pip install psycopg2
+```
+
+Enable the database monitoring propagation feature by setting the following enviornment variable:
+   `DD_TRACE_SQL_COMMENT_INJECTION_MODE=full`
```
Comment (mabdinur) [inline content/en/database_monitoring/guide/connect_dbm_and_apm.md:160]:
```suggestion
   - `DD_TRACE_SQL_COMMENT_INJECTION_MODE=full`
```

File: content/en/database_monitoring/guide/connect_dbm_and_apm.md [hunk 3]
```diff
@@ -144,6 +144,53 @@ client.query("SELECT 1;")
...
+```
+pip install psycopg2
+```
+
+Enable the database monitoring propagation feature by setting the following enviornment variable:
+   - `DD_TRACE_SQL_COMMENT_INJECTION_MODE=full`
+
+For the best user experience ensure the following enviornemnts variables are set in your application:
+   - `DD_SERVICE=(application name)`
+   - `DD_ENV=(application enviornment)`
```
Comment (alexandre-normand) [inline content/en/database_monitoring/guide/connect_dbm_and_apm.md:164]:
```suggestion
   - `DD_ENV=(application environment)`
```

**Review by alexandre-normand — COMMENTED** [review]
Can you also add the python tracer to the supported tracers list in the [before you begin](https://github.com/DataDog/documentation/blob/7b02ad8128195b48ae2c81181e04b92ff67c5100/content/en/database_monitoring/guide/connect_dbm_and_apm.md#before-you-begin) header section?

**Comment by alexandre-normand** [comment]
From my end (database monitoring), this looks great. Thanks, @mabdinur ! 🙇

File: content/en/database_monitoring/guide/connect_dbm_and_apm.md [hunk 4]
```diff
@@ -145,6 +146,51 @@ client.query("SELECT 1;")
...
+conn = psycopg2.connect(**POSTGRES_CONFIG)
+cursor = conn.cursor()
+# execute sql queries
+cursor.execute("select 'blah'")
+cursor.executemany("select %s", (("foo",), ("bar",)))
+```
+
+[1]: https://ddtrace.readthedocs.io/en/stable/release_notes.html
+[2]: https://ddtrace.readthedocs.io/en/stable/integrations.html#module-ddtrace.contrib.psycopg
+
```
Comment (mabdinur) [inline content/en/database_monitoring/guide/connect_dbm_and_apm.md:193]:
```suggestion

{{% /tab %}}

```

File: content/en/database_monitoring/guide/connect_dbm_and_apm.md [hunk 5]
```diff
@@ -206,5 +254,7 @@ client.query('SELECT $1::text as message', ['Hello world!'], (err, result) => {
 [6]: https://github.com/dataDog/dd-trace-rb
 [7]: https://github.com/brianmario/mysql2
 [8]: https://github.com/ged/ruby-pg
-[9]: https://github.com/DataDog/dd-trace-js
-[10]: https://node-postgres.com/
+[9]: https://github.com/dataDog/dd-trace-py
+[10]: https://www.psycopg.org/docs/index.html
+[11]: https://github.com/DataDog/dd-trace-js
+[12]: https://node-postgres.com/
```
Comment (mabdinur) [inline content/en/database_monitoring/guide/connect_dbm_and_apm.md:260]:
```suggestion
[9]: https://github.com/DataDog/dd-trace-js
[10]: https://node-postgres.com/
[11]: https://github.com/dataDog/dd-trace-py
[12]: https://www.psycopg.org/docs/index.html
```

File: content/en/database_monitoring/guide/connect_dbm_and_apm.md [hunk 6]
```diff
@@ -17,6 +17,7 @@ Supported tracers
 : [dd-trace-go][3] >= 1.44.0 (support for [database/sql][4] and [sqlx][5] packages)<br />
 [dd-trace-rb][6] >= 1.6.0 (support for [mysql2][7] and [pg][8] gems)<br />
 [dd-trace-js][9] >= 3.9.0 or >= 2.22.0 (support for [postgres client][10])
```
Comment (maycmlee) [inline content/en/database_monitoring/guide/connect_dbm_and_apm.md:19]:
```suggestion
[dd-trace-js][9] >= 3.9.0 or >= 2.22.0 (support for [postgres client][10])<br />
```

File: content/en/database_monitoring/guide/connect_dbm_and_apm.md [hunk 7]
```diff
@@ -208,3 +256,5 @@ client.query('SELECT $1::text as message', ['Hello world!'], (err, result) => {
 [8]: https://github.com/ged/ruby-pg
 [9]: https://github.com/DataDog/dd-trace-js
 [10]: https://node-postgres.com/
+[11]: https://github.com/dataDog/dd-trace-py
```
Comment (alexandre-normand) [inline content/en/database_monitoring/guide/connect_dbm_and_apm.md:259]:
```suggestion
[11]: https://github.com/DataDog/dd-trace-py
```

File: content/en/database_monitoring/guide/connect_dbm_and_apm.md [hunk 8]
```diff
@@ -145,6 +146,53 @@ client.query("SELECT 1;")
...
+   - `DD_SERVICE=(application name)`
+   - `DD_ENV=(application environment)`
+   - `DD_VERSION=(application version)`
+
+Full example:
+```python
+
+import psycopg2
+
+#TODO: update postgres configurations
```
Comment (alexandre-normand) [inline content/en/database_monitoring/guide/connect_dbm_and_apm.md:174]: Is the TODO meant for the end-user copying that example?

File: content/en/database_monitoring/guide/connect_dbm_and_apm.md [hunk 9]
```diff
@@ -145,6 +146,53 @@ client.query("SELECT 1;")
...
 {{% /tab %}}
 
+{{% tab "Python" %}}
+
+Update your app dependencies to include [dd-trace-py>=1.7.0][1]:
+```
+pip install "ddtrace>=1.7.0"
+```
+
+Install [psycopg2][2] (Note - DBM-APM linking is not currently supported for mysql clients):
```
Comment (maycmlee) [inline content/en/database_monitoring/guide/connect_dbm_and_apm.md:156]:
```suggestion
Install [psycopg2][2] (**Note**: Connecting DBM and APM is not supported for MySQL clients):
```

File: content/en/database_monitoring/guide/connect_dbm_and_apm.md (hunk 8 above)
Comment (mabdinur) [reply content/en/database_monitoring/guide/connect_dbm_and_apm.md:174]: Yes

**Review by alexandre-normand — APPROVED** [review]

**Review by maycmlee — APPROVED** [review]
@mabdinur LGTM! Let me know if I can go ahead and merge.

**Comment by mabdinur** [comment]
> @mabdinur LGTM! Let me know if I can go ahead and merge.

I noticed the linting and link preview checks are failing. Once those are passing then this PR is good to go 🥳

**Comment by alexandre-normand** [comment]
@maycmlee : We'd like to have this documentation update deployed so we can use it when reaching out to trial customers. I tried to merge master to see if this would help with the CI failing on the linting job but that didn't help. Is there a known issue that is causing this?

**Comment by maycmlee** [comment]
@alexandre-normand Yes, unfortunately, the linting job is flakey and our link checker test has also been flakey recently. Thanks for merging master in, I'll take a look once all the tests have passed, and will either fix it or merge the PR if it looks good to me.

**Pull request merged**